converter/src/DelphesEDM4HepConverter.cc

```diff
@@ -269,12 +281,23 @@ namespace k4SimDelphes {
       // that would probably make the matching that is done below much harder
 
       auto       cand     = particleCollection->create();
-      const auto momentum = delphesCand->P4();  // NOTE: assuming massless here!
+      // NOTE: Delphes assumes m=0 for photons and m=0.497611 for neutral hadrons (KL)
+      const auto momentum = delphesCand->P4();
       // TODO: fill this when it is available later, when when we link the references?
       // cand.setCharge(delphesCand->Charge);
       cand.setMomentum({(float)momentum.Px(), (float)momentum.Py(), (float)momentum.Pz()});
       cand.setEnergy(delphesCand->E);
-
+      cand.setMass(delphesCand->P4().M());
+      // NOTE: Particle-Flow Neutral are either photons or K_L in Delphes
```

Comment:
Is there any chance that we get non-neutral candidates here? In principle those should all go to the `Tracks`, right? I.e. `Tower` will always be a neutral?

Reply:
It is possible for a charged hadron/electron to  fall in the `EFlowNeutralHadron`/`EFlowPhoton` collections and form a `Tower`, but in that case Delphes simply creates a K_L or a Photon.